core/src/executor/evaluate/expr.rs

```diff
@@ -85,6 +93,10 @@ pub fn between<'a>(
     low: Evaluated<'a>,
     high: Evaluated<'a>,
 ) -> Result<Evaluated<'a>> {
+    if target.is_null() {
+        return Ok(target);
+    }
```

Comment:
we may not need these `target.is_null()` check.
currently only the issue is from here.

```rust
   UnaryOperator::Not => v
            .try_into()
            .map(|v: bool| Evaluated::Value(Value::Bool(!v))),
```

for the other cases, null value is already being handled properly.
could we try change `Not` handler to return Null on null input?

Reply:
Well... I'm not sure that I understood well but I think the `target.is_null()` check in `fn between` is still needed because `NULL BETWEEN <lhs> AND <rhs>` should return `NULL`.

Reply:
I separated the part to `.unary_not()` function like other unary operators (e.g., `.unary_plus()`).